File: content/5.polympiads2/1.introduction-aux-arbres/.codes/solution.py
nb_noeuds = int(input())
parents = list(map(int, input().split()))

# On compte pour chaque noeud son nombre de feuille
nombre_enfants = [0 for i in range(nb_noeuds)] # on créé une liste de compteur

for i_noeud in range(1, nb_noeuds):
	# On incrémente le compteur du parent
	nombre_enfants[parents[i_noeud]] += 1

# On compte le nombre de feuilles
nombre_feuilles = 0
for i_noeud in range(nb_noeuds):
	if (nombre_enfants[i_noeud] == 0):
		nombre_feuilles += 1
	
print(nombre_feuilles)

File: content/5.polympiads2/1.introduction-aux-arbres/.codes/solution.java
import java.util.Scanner;

class Main { 
	public static void main (String[] args) {
		Scanner sc = new Scanner(System.in);
		int nbNoeuds = sc.nextInt();

		int[] parents = new int[nbNoeuds];
		for (int i = 0; i < nbNoeuds; i ++) {
			parents[i] = sc.nextInt();
		}
		
		// On compte pour chaque noeud son nombre de feuille
		int[] nombreEnfants = new int[nbNoeuds]; // on créé une liste de compteur
		for (int iNoeud = 1; iNoeud < nbNoeuds; iNoeud ++) {
			// On incrémente le compteur du parent
			nombreEnfants[parents[iNoeud]] ++;
		}
	
		// On compte le nombre de feuilles
		int nombreFeuilles = 0;
		for (int iNoeud = 0; iNoeud < nbNoeuds; iNoeud ++) {
			if (nombreEnfants[iNoeud] == 0) {
				nombreFeuilles ++;
			}
		}
	
		System.out.println(nombreFeuilles);
	}
}